### DISCUSSION TITLE
Autograder 没有报错，测评自动停止

### DISCUSSION THREAD

**Discussion by Zhou7D663** [arcsysu/SYsU-lang#108] [Q&A]
我提交了三次，使用同一个作业文件，都自动停止测评了，且三次的得分有不一样的：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/c655d066-36f1-4a17-acdc-df3a08a48074)
测试报告中没有报错，运行日志戛然而止：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/a813fe3f-1196-4a36-96ac-e9998ab2d83a)
未测试的样例无法查看具体信息：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/84aa521e-f3b6-4118-9d49-dca7dcff49ce)

**Comment by xuantengh** [comment]
在本地使用 docker 进行测试的结果呢?
在 `.tar.gz` 文件所在的目录运行: 
```bash
docker run --name sysu-lang-test -v `pwd`:/autograder/submission wukan0621/sysu-lang
```

**Reply by Zhou7D663** [reply]
在docker的环境中，测试到[514/647]时出现了问题，测试突然停止，与下面两次情况相同：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/5791eb0f-a47c-46b0-b46a-5d6cd38a86c0)

**Reply by xuantengh** [reply]
> 在docker的环境中，测试到[514/647]时出现了问题，测试突然停止

有具体报错信息吗? 还是容器直接退出了?

**Reply by xuantengh** [reply]
怀疑很有可能是爆内存了, 可以在 docker 命令中添加 `--memory=10g --cpus=1` 选项试试?

**Reply by Zhou7D663** [reply]
我单独对出错的文件进行了中间代码生成的测试，结果为段错误Segmentation fault，无法生成中间代码

**Reply by Zhou7D663** [reply]
docker中没看到啥报错信息，只有warning

**Reply by Zhou7D663** [reply]
段错误问题解决
SYsU-lang-tester-perfermance/performance_test2021-public/conv0.sysu.c的memmove函数是一个库函数：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/00302803-d051-4525-8c0b-456d3fb05a52)
clang生成的json格式的抽象语法树中，对于此函数会自动生成一个FunctionDecl：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/392ac081-f4e3-44e3-94e9-2a3b795a87e8)
而将c文件中的memmove当作库函数的重新声明：
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/b442bb73-56e4-4b42-a8a4-1ed8d9b1fe1a)
![image](https://github.com/arcsysu/SYsU-lang/assets/125145652/3aca2b67-870f-435e-980a-3d9223066509)
而我的main.cc中对于函数的重新声明的一些处理最终导致了段错误。
